hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -146,6 +152,16 @@ public void setPipelineProvider(ReplicationType replicationType,
     pipelineFactory.setProvider(replicationType, provider);
   }
 
+  @VisibleForTesting
+  public void setAllowPipelineCreation(boolean newState) {
+    this.allowPipelineCreation.set(newState);
+  }
+
+  @VisibleForTesting
+  public boolean getAllowPipelineCreation() {
+    return allowPipelineCreation.get();
+  }
+
```

Comment:
I would suggest to use pipelineCreationAllowed as the internal state name, it makes the code easier to read out as I I see.

I know this is a regular getter and and setter pair, but...
In the tests, the set method is always called with true as the parameter, so I suggest to call this method simply allowPipelineCreation() without a parameter, the getter part of the pair in this case can be called as isPipelineCreationAllowed().

This suggestion is simply about expressions from the usage point of view, and I don't have a strong opinion on this, however if we want to keep this name and want to name it as a regular getter setter as it is suggested by the books then we should use isAllowPipelineCreation as the getter name.

Reply:
I think your naming is better. I will push a new commit for that change.